plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -799,6 +796,28 @@ public function build_frontend_query( $query, $block, $page ) {
 		);
 	}
 
+	/**
+	 * Return a custom query based on attributes, filters and global WP_Query.
+	 *
+	 * @param array $parsed_block  A representative array of the block being rendered. See WP_Block_Parser_Block
+	 *
+	 * @return array
+	 */
+	public function set_fallback_query_id( $parsed_block ) {
+
+		$block_name = $parsed_block['blockName'] ?? null;
+
+		if ( 'woocommerce/product-collection' !== $block_name ) {
+			return $parsed_block;
+		}
+
+		if ( empty ( $parsed_block['attrs']['queryId'] ) ) {
+			$parsed_block['attrs']['queryId'] = rand();
+		}
```

Comment:
Using `rand()` to generate a queryId creates a critical pagination issue. Since queryId is used for pagination purposes, we don't want it to change on every page load. However, with `rand()`, that's exactly what happens - it generates a new random queryId each time the page loads.

I've tested this and confirmed the issue:
- Create a new post with a Product Collection block
- Remove the `queryId` attribute in Code Editor
- Publish and view the post
- Try navigating to page 2 of the collection

The pagination breaks because each attempt generates a new random queryId. For example, after trying to navigate to page 2 a couple of times, the URL looks like this:
`http://mstore.studio/245-2/?query-225930939-page=2&query-1382933995-page=2`. And I am still on page 1. Are you able to reproduce this?

Reply:
Great catch! Not sure what to suggest here. What if we create a hash from the query args (Except page)?

Reply:
ps - where are you doing `Remove the queryId attribute in Code Editor` ?

Reply:
> ps - where are you doing Remove the queryId attribute in Code Editor ?

Oh! Sorry for not being clear. Here is a quick demo video of what I meant:

https://github.com/user-attachments/assets/3908f7bc-1948-437b-a48c-4929bb769826